packages/core-data/src/selectors.js

```diff
@@ -233,16 +236,13 @@ export const getEntityRecords = createSelector(
 		// assigned for the given parameters, then it is known to not exist. Thus, a
 		// return value of an empty array is used instead of `null` (where `null` is
 		// otherwise used to represent an unknown state).
-		const queriedState = get( state.entities.data, [
-			kind,
-			name,
-			'queriedData',
-		] );
+		const queriedState = getQueriedState( state, kind, name );
 		if ( ! queriedState ) {
 			return [];
 		}
 		return getQueriedItems( queriedState, query );
-	}
+	},
+	( state, kind, name ) => [ getQueriedState( state, kind, name ) ]
```

Comment:
Will we have the same result if we use a const array? If it's the case, this seems to be way harder to understand and also more costly in terms of performance than just using a const.

Reply:
Good point. I don't think that there's a difference in this specific case. However, I'd still prefer to use a more general solution so that future changes won't accidentally break the memoization.

Let's use a const array here, as it's more simpler and performant. I still think it's too complicated to choose the right memoization method with the current approach though 🤔 .

Reply:
Yes, I agree that memoization is hard in general. Redux forces us into thinking more about these details in a way.